luarules/gadgets/unit_dgun_behaviour.lua

```diff
@@ -79,13 +80,18 @@ end
 
 function gadget:UnitPreDamaged(unitID, unitDefID, unitTeam, damage, paralyzer, weaponDefID, projectileID, attackerID, attackerDefID, attackerTeam)
 	if dgunDamages[weaponDefID] and isCommander[unitDefID] and isCommander[attackerDefID] then
-		damagedUnits[projectileID] = damagedUnits[projectileID] or {}
-		if damagedUnits[projectileID][unitID] then
-			return 0
-		else
-			damagedUnits[projectileID][unitID] = true
-			return dgunDamages[weaponDefID][unitArmorType[unitDefID]]
-		end
+--		damagedUnits[projectileID] = damagedUnits[projectileID] or {}
+	--	if damagedUnits[projectileID][unitID] then
+	--		return 0
+--		else
+	--		damagedUnits[projectileID][unitID] = true
+	--		return dgunDamages[weaponDefID][unitArmorType[unitDefID]]
+--		end
+--	if isDGun[weaponDefID] and isCommander[unitDefID] and isCommander[attackerDefID] then
+		Spring.DeleteProjectile(projectileID)
+		local XPos, YPos, ZPos = Spring.GetUnitPosition(unitID)		
+		Spring.SpawnCEG("commander-spawn-explo", XPos, YPos, ZPos,0,0,0,0,0)
```

Comment:
```suggestion
		Spring.SpawnCEG("commander-spawn-explo", XPos, YPos, ZPos,0,0,0,0,0)\
		return 0
```
The function still needs to return the damage value to apply.

Reply:
Huh, odd, it worked ingame for me, but thanks.